domain/port/state/types.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import "github.com/juju/juju/core/network"
+
+// protocol represents a network protocol type and its ID in DQLite.
+type protocol struct {
+	ID   int    `db:"id"`
+	Name string `db:"protocol"`
+}
+
+// portRange represents a range of ports for a given protocol.
+type portRange struct {
+	Protocol string `db:"protocol"`
+	FromPort int    `db:"from_port"`
+	ToPort   int    `db:"to_port"`
+}
+
+func (pr portRange) decode() network.PortRange {
+	return network.PortRange{
+		Protocol: pr.Protocol,
+		FromPort: pr.FromPort,
+		ToPort:   pr.ToPort,
+	}
+}
+
+// endpointPortRange represents a range of ports for a give protocol for a
+// given endpoint.
+type endpointPortRange struct {
+	PrococolId int    `db:"protocol_id"`
```

Comment:
Got a type here.

Reply:
This field shouldn't even be included. Good spot